xgettext.js

```diff
@@ -123,30 +131,29 @@ XGettext.prototype._normalizeKeyword = function( keyword ) {
  */
 XGettext.prototype._parseInput = function( input ) {
 	var comments = [],
-		parseOptions = { locations: true },
+		parseOptions = this.options.parseOptions,
 		ast;
 
+	ast = parser.parse( input, parseOptions );
+
 	if ( typeof this.options.commentPrefix !== 'undefined' ) {
 		// Optionally locate translator comments
 		var rxCommentMatch = new RegExp( '^\\s*' + this.options.commentPrefix, 'i' );
-
-		parseOptions.onComment = function( block, text, start, end, line ) {
+		ast.comments.forEach( function( comment) {
+			var text = comment.value;
 			var isTranslatorComment = rxCommentMatch.test( text );
 
 			if ( isTranslatorComment ) {
 				comments.push({
 					value: text.replace( rxCommentMatch, '' ).trim(),
-					line: line
+					line: comment.loc.start
 				});
 			}
-		};
+		});
```

Comment:
Same here. FYI the coding standard here is currently more like the [WordPress JS Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/) (no spaces around function arg), but ideally moving to eventually adopting [`eslint-config-wpcalypso`](https://github.com/Automattic/eslint-config-wpcalypso)